proxy/api/src/project.rs

```diff
@@ -71,6 +75,76 @@ where
     }
 }
 
+/// A Radicle project that you're interested in but haven't contributed to.
+///
+/// See [`Projects`] for a detailed breakdown of both kinds of projects.
+#[derive(Serialize)]
+pub struct Tracked(Project);
+
+impl Deref for Tracked {
+    type Target = Project;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A user can view projects that are in different states in their application.
+pub struct Projects {
+    /// A project that is tracked is one that the user has replicated onto their device but has not
+    /// made any changes to. A project is still considered tracked if they checked out a working copy but
+    /// have not performed any commits to the references.
+    pub tracked: Vec<Tracked>,
+    /// A project that is "mine" is one that the user has either:
+    ///     a. Created themselves using the application.
+    ///     b. Replicated (see tracked above), checked out a working copy, and push changes to
+    ///     references.
+    ///
+    /// The conditions imply that a project is "mine" if I am the maintainer or I have contributed
+    /// to the project.
+    pub mine: Vec<Project>,
+}
+
+impl Projects {
+    /// List all the projects that are located on your device. These projects could either be "tracked"
+    /// or "mine".
+    ///
+    /// See [`Projects`] for a detailed breakdown of both kinds of projects.
+    ///
+    /// # Errors
+    ///
+    ///   * We couldn't get the list of projects
+    ///   * We couldn't inspect the `signed_refs` of the project
+    ///   * We couldn't get stats for a project
+    pub fn list(api: &coco::Api) -> Result<Self, error::Error> {
+        let mut projects = Self {
+            tracked: vec![],
+            mine: vec![],
+        };
+        for project in api.list_projects()? {
+            let refs = api.list_project_refs(&project.urn())?;
+            let project = api.with_browser(&project.urn(), |browser| {
+                let stats = browser.get_stats().map_err(coco::Error::from)?;
+                Ok((project, stats).into())
+            })?;
+            if refs.heads.is_empty() {
+                projects.tracked.push(Tracked(project))
+            } else {
+                projects.mine.push(project)
+            }
+        }
+
+        Ok(projects)
+    }
+
+    /// Give back an iterator over all the [`Project`]s.
+    pub fn into_iter(self) -> impl Iterator<Item = Project> {
```

Comment:
Shouldn't we impl [`IntoIterator`](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html) if we want this kind of API?